man/fi_mrail.7.md

```diff
@@ -80,6 +80,12 @@ The ofi_mrail provider checks for the following environment variables.
 *FI_OFI_MRAIL_ADDR_STRC*
 : Comma delimited list of individual rail addresses in FI_ADDR_STR format.
 
+*FI_OFI_MRAIL_STRIPING_THRESHOLD*
+: Minimal message size to allow striping (default: 16385).
+
+*FI_OFI_MRAIL_DEFAULT_RAIL*
+: Rail to use when message is not striped (<0 means round-robin, default: 0).
```

Comment:
This is mixing the striping algorithm with the default rail selection.  I would separate them.  For the striping algorithm, there are several options.  This proposal is already introducing different algorithms based on size.  RxM has 3 transfer algorithms based on size, so I can envision mrail ending up with multiple algorithms as well.

The question is: how do we best expose the selection choice(s) to the user? Do we have a single MRAIL_ALGORITHM (e.g. round-robin striping, round-robin message, least busy, separate msg and rma, blah blah)?  Or do we specify these per message size?  Per type of operation?

Environment variables are nearly an API, so we should figure out what gives us the most flexibility without insanity.

Reply:
At least we need two ranges of message sizes (small, large) that may use different algorithm. We have the option to add a third one (medium) in the future. The algorithms can starts with "fixed", "round-robin", "round-robin striping" and can be augmented in the future with new options.